src/modules/roc_core/target_libunwind/roc_core/backtrace.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#define UNW_LOCAL_ONLY
+#include <libunwind.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "roc_core/backtrace.h"
+
+/* Function to convert unsigned int to decimal/hex string.
+ * This function is signal safe.
+ * Parameters :
+ * number = the number to be converted.
+ * base = 10 for decimal to char* and 16 for hex to char* conversion.
+ * output_string = the decimal/hex string
+ */
+
+namespace {
+
+enum { MaxDigits = 10, MaxLen = 200 };
+
+void uint32_to_string(uint32_t number, uint32_t base, char* output_string) {
+    int i = MaxDigits;
+    int j = 0;
+    do {
+        output_string[i] = "0123456789abcdef"[number % base];
+        i--;
+        number = number / base;
+    } while (number > 0);
+
+    while (++i < MaxDigits + 1)
+        output_string[j++] = output_string[i];
+    output_string[j] = 0;
+}
+
+/* Checking if backtrace is empty or not.
+ * This must be signal safe.
+ */
+bool is_backtrace_available() {
+    /* To store the snapshot of the CPU registers.
+     */
+    unw_context_t context;
+
+    /* To get snapshot of the CPU register.
+     * unw_getcontext() is signal safe.
+     */
+    if (unw_getcontext(&context) < 0)
+        return false;
+
+    /* To point to the current frame in the call stack.
+ */
```

Comment:
Nitpick: bad comment formatting here and below (if it's caused by clang-format, maybe you're using an outdated version? I'm using 7.1.0.)